daffodil-test/src/test/resources/org/apache/daffodil/section07/discriminators/nestedChoiceDiscriminator.tdml

```diff
@@ -0,0 +1,306 @@
+
+                        This is not necessarily desirable. The reason why the message failed is masked by this.
+                        -->
+                        <xs:choice>
+                          <xs:element name="message2" type="ex:messageType"/>
+                          <xs:group ref="ex:messageParseError"/>
+                        </xs:choice>
+                      </xs:sequence>
+                    </xs:choice>
+                    <xs:sequence>
+                      <!--
+                      default case for completely unrecognized message
+                      -->
+                      <xs:sequence>
+                        <xs:annotation>
+                          <xs:appinfo source="http://www.ogf.org/dfdl/">
+                            <!--
+                            If capture variable not set, then just fail here, and report an error.
+                            -->
+                            <dfdl:discriminator message='{ fn:concat("Unrecognized messageID: ", ./messageID) }'
+                                                test='{ $ex:capture }'/>
+                          </xs:appinfo>
+                        </xs:annotation>
+                      </xs:sequence>
+                      <!--
+                       Otherwise, create this unrecognized capture element.
+
+                       Notice that maxOccurs="0" so valid data never contains these.
+                       Validation checking will reject data that contains these.
+
+                       This tells us that the same DFDL schema, with these capture elements expressed in it
+                       can be used for both parse/unparse and for separate validation of the data.
+                       -->
+                      <xs:element name="unrecognizedMessage" minOccurs="0" maxOccurs="0"
+                                  dfdl:occursCountKind="expression"
+                                  dfdl:occursCount='{ if ($ex:capture) then 1 else 0 }'
+                                  type="xs:string"/>
+                    </xs:sequence>
+                  </xs:choice>
+                </xs:sequence>
+              </xs:sequence>
+            </xs:complexType>
+          </xs:element>
+        </xs:sequence>
+      </xs:complexType>
+    </xs:element>
+
+    <xs:group name="messageParseError">
+      <xs:sequence>
+        <!-- default case for failure to parse a message -->
+        <xs:sequence>
+          <xs:annotation>
+            <xs:appinfo source="http://www.ogf.org/dfdl/">
+              <!--
+              If capture variable not set, then just fail here, and report an error.
+              -->
+              <dfdl:discriminator message='{ fn:concat("Unable to parse message with messageID: ", ./messageID) }'
+                                  test='{ $ex:capture }'/>
+            </xs:appinfo>
+          </xs:annotation>
+        </xs:sequence>
+        <!--
+         Otherwise, create this unrecognized capture element
```

Comment:
Not unrecognized capture element. This is a message parse error capture element.